UITests/README.md

```diff
@@ -1,29 +1,73 @@
+1. Generate the test signing certificate:
 
-- Windows 10 PC with the latest Windows 10 version (Version 1809 or later)
-- Microsoft Visual Studio 2017 or later
-- [Xaml Controls Gallery](../../XamlControlsGallery/XamlControlsGallery.sln) built and deployed 
+```shell
+    PS> .\build\GenerateTestPfx.ps1
+```
 
+1. Build and publish the Xaml Controls Gallery from the command line, e.g.:
 
-## Getting Started
+```shell
+    dotnet.exe publish WinUIGallery\WinUIGallery.sln /p:AppxPackageDir=AppxPackages\ /p:platform=x64 /p:PublishProfile=./WinUIGallery/Properties/PublishProfiles/win10-x64.pubxml
+```
 
-1. Open `UITests.sln` in Visual Studio
-2. Select **Test** > **Windows** > **Test Explorer**
-3. Select **Run All** on the test pane or through menu **Test** > **Run** > **All Tests**
+1. Locate the Xaml Controls Gallery package output folder from above and deploy for testing:
 
-> Once the project is successfully built, you can use the **TestExplorer** to pick and choose the test scenario(s) to run
+```shell
+    PS> .\WinUIGallery\AppxPackages\WinUIGallery.Desktop_Test\Install.ps1
+```
 
+## Install and execute WinAppDriver
 
-## Adding/Updating Test Scenario
+1. Download and install the latest version of WinAppDriver from [here](https://github.com/microsoft/WinAppDriver/releases)
+
+1. Run WinAppDriver (required for running tests on cmdline or in VS):
+
+```shell
+    C:\Program Files (x86)\Windows Application Driver>WinAppDriver.exe
+```
+
+## Build and run/debug Tests
+
+1. Build UITests:
+
+```shell
+    >msbuild UITests\UITests.sln
+```
+
+1. Run test cases built above on command line:
+
+```shell
+    vstest.console.exe D:\git\WinUI-Gallery\UITests\bin\x64\Debug\net7.0\UITests.dll
+```
+
+1. Debug test cases in Visual Studio
+   * Open `UITests.sln` in Visual Studio
+   * Select **Test** > **Windows** > **Test Explorer**
+   * Select **Run All** on the test pane or through menu **Test** > **Run** > **All Tests**
+
+   Once the project is successfully built, you can use the **TestExplorer** to pick and choose the test scenario(s) to run/debug
+
+## Test output
+
+In either scenario above (cmdline or VS), if the tests run successfully, you should see:
+
+* WinAppDriver console spew indicating successful launch of the test app
+* The Xaml Controls Gallery launch and run with automated UI interactions
+* vstest.console.exe spew (if cmdline) showing test case status
+
+
+## Adding/Updating Test Scenarios
 
 Please follow the guidelines below to maintain test reliability and reduce test execution time:
+
 1. Provide a complete set of interactions (if applicable) for each new control
-2. Aim for simple and reliable scenario using the least amount of test steps
-3. Reuse existing application session when possible to reduce unnecessary application re-launching
+1. Aim for simple and reliable scenario using the least amount of test steps
+1. Reuse existing application session when possible to reduce unnecessary application re-launching
```

Comment:
Not sure if this still applies since it would be quite an effort to not reuse the existing application session.